packages/stream-object/src/index.ts

```diff
@@ -33,6 +63,35 @@ export const streamObject = async <T extends Schema>(options: StreamObjectOption
     schemaDescription: undefined,
     schemaName: undefined,
   }).then(({ chunkStream, stepStream, textStream: rawTextStream }) => {
+    let elementStream: ReadableStream<Infer<T>> | undefined
+    let index = 0
+
+    if (options.output === 'array') {
+      let [textStream, rawTextStream] = rawTextStream.tee()
```

Comment:
```suggestion
      let [textStream, rawElementStream] = rawTextStream.tee()
```

Reply:
reassigning the rawTextStream is to avoid affecting the objectstream below
but it is reasonable if output: array will make no objectstream return